packages/plugin-js-packages/src/lib/runner/audit/constants.ts

```diff
@@ -28,9 +30,41 @@ export const normalizeAuditMapper: Record<
   npm: npmToAuditResult,
   'yarn-classic': yarnv1ToAuditResult,
   'yarn-modern': yarnv2ToAuditResult,
-  pnpm: () => {
-    throw new Error('PNPM audit is not supported yet.');
-  },
+  pnpm: pnpmToAuditResult,
+};
+
+const filterNpmAuditResults = (
+  results: Record<DependencyGroup, AuditResult>,
+) => ({
+  prod: results.prod,
+  dev: filterAuditResult(results.dev, 'name', results.prod),
+  optional: filterAuditResult(results.optional, 'name', results.prod),
+});
+
+const filterPnpmAuditResults = (
+  results: Record<DependencyGroup, AuditResult>,
+) => ({
+  prod: results.prod,
+  dev: results.dev,
+  optional: filterAuditResult(
+    filterAuditResult(results.optional, 'id', results.prod),
+    'id',
+    results.dev,
+  ),
+});
+
+export const postProcessingAuditMapper: Record<
+  PackageManager,
+  (
+    result: Record<DependencyGroup, AuditResult>,
+  ) => Record<DependencyGroup, AuditResult>
+> = {
+  npm: filterNpmAuditResults, // prod dependencies need to be filtered out manually since v10
+  /* eslint-disable @typescript-eslint/naming-convention */
+  'yarn-classic': results => results,
+  'yarn-modern': results => results,
+  /* eslint-enable @typescript-eslint/naming-convention */
```

Comment:
These could be omitted if they don't need any post-processing.

Reply:
Originally, I had a ternary operator which provided only for those that have it. If I omit it here then the `Record` will break as not all package managers are listed. Should I return to the direct ternary operator? 🤔

Reply:
Partial Record added.